consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -308,21 +308,26 @@ private void cancelAnyExistingBlockCreationTasks(final PayloadIdentifier payload
           "New payloadId {} received so cancelling block creation tasks for the following payloadIds: {}",
           payloadIdentifier,
           existingPayloadIdsBeingBuilt);
-      blockCreationTask.values().forEach(BlockCreationTask::cancel);
+
+      blockCreationTask.keySet().forEach(this::cleanupBlockCreationTask);
     }
   }
 
-  @Override
-  public void finalizeProposalById(final PayloadIdentifier payloadId) {
-    LOG.debug("Finalizing block proposal for payload id {}", payloadId);
+  private void cleanupBlockCreationTask(final PayloadIdentifier payloadIdentifier) {
     blockCreationTask.computeIfPresent(
-        payloadId,
+        payloadIdentifier,
         (pid, blockCreationTask) -> {
           blockCreationTask.cancel();
-          return blockCreationTask;
+          return null;
         });
   }
 
+  @Override
+  public void finalizeProposalById(final PayloadIdentifier payloadId) {
+    LOG.debug("Finalizing block proposal for payload id {}", payloadId);
```

Comment:
Is this going to spam the logs? Most of the time I think we would cancel the block creation as we are constantly trying to rebuild a better one.

Reply:
I haven't changed anything with this, just refactored a bit. This is logged every time engine_getPayload is called, it's the happy path.